### PR TITLE
gnrc_tcp: rewrite passive open

### LINKED ISSUE #10664: TCP Sockets can not be used / built
#### Description
Can not build code using TCP sockets by writing `USEMODULE += gnrc_sock_tcp`.
I assume part of the probleme might be missing implementation? As you can see in [1] there only is a directory for UDP (which i successfully used previously) but not TCP. Since the documentation [2] describes the use of TCP by example and explicitly states 

> First you need to include a module that implements this API in your application's Makefile. For example the implementation for GNRC is called gnrc_sock_tcp.

i considered this a bug (as opposed to a feature request).

#### Steps to reproduce the issue
Try to build a project around TCP sockets, for example as described in the documentation here [2]

#### Expected results
Successfull Compilation

#### Actual results
Compiler Error: 
`arm-none-eabi-gcc: error: <path-redacted>/bin/cc2538dk/gnrc_sock_tcp.a: No such file or directory`

#### Versions
```
Operating System Environment
-----------------------------
       Operating System: "Arch Linux" 
                 Kernel: Linux 4.18.16-arch1-1-ARCH x86_64 unknown

Installed compiler toolchains
-----------------------------
             native gcc: gcc (GCC) 8.2.1 20180831
      arm-none-eabi-gcc: arm-none-eabi-gcc (Arch Repository) 8.2.0
                avr-gcc: avr-gcc (GCC) 8.2.0
       mips-mti-elf-gcc: missing
             msp430-gcc: missing
   riscv-none-embed-gcc: missing
                  clang: clang version 7.0.0 (tags/RELEASE_700/final)

Installed compiler libs
-----------------------
   arm-none-eabi-newlib: "3.0.0"
    mips-mti-elf-newlib: missing
riscv-none-embed-newlib: missing
               avr-libc: "2.0.0" ("20150208")

Installed development tools
---------------------------
                  cmake: cmake version 3.12.4
               cppcheck: missing
                doxygen: 1.8.14
                 flake8: missing
                    git: git version 2.19.1
                   make: GNU Make 4.2.1
                openocd: Open On-Chip Debugger 0.10.0
                 python: Python 3.7.1
                python2: Python 2.7.15
                python3: Python 3.7.1
             coccinelle: missing
```
#### References
[1] https://github.com/RIOT-OS/RIOT/tree/master/sys/net/gnrc/sock
[2] https://riot-os.org/api/group__net__sock__tcp.html#ga0144778bc074e291fcb0d43bd35c0179

### PR DESCRIPTION
**Pull request by brummer-simon**
### Contribution description
This PR rewrites the passive connection establishment mechanism of GNRC_TCP.
The new version aligns the GNRC_TCP interface with the SOCK_TCP interface with the goal to make a upcoming SOCK integration straight forward and trivial.

Since this is a large change, I've split the PR into three commits (api, implementation and tests) to help reviewing this.
The new tests are written in different style than existing tests, please ignore the difference in style for now, I plan to align the existing  test structure in a dedicated PR on after merging this. 

### Testing procedure
This PR is covered by the GNRC_TCP test suite.
To test this PR follow the instructions under tests/gnrc_tcp/Readme.md

### Issues/PRs references
Fixes #10664 - TCP Sockets can not be used / built

### CODE REVIEW

**Comment by brummer-simon** [comment]
@miri64 - it doesn't fix #10664 but its one of the last steps before fixing it.

**Comment by MrKevinWeiss** [comment]
Should we move this to next release as it is soft feature freeze and this contains an API change?

**Comment by brummer-simon** [comment]
This PR and the followup PR contain large API changes. Although the changes are tested in depth, it might be much effort to review all of this. I personally don't care if this is part of this release or the next. From my side, do whatever relaxes your release management. ;)

File: sys/net/gnrc/transport_layer/tcp/gnrc_tcp_fsm.c
Comment (benpicco) [inline sys/net/gnrc/transport_layer/tcp/gnrc_tcp_fsm.c:216]:
```suggestion
                _gnrc_tcp_eventloop_sched(&tcb->event_timeout,
                                          CONFIG_GNRC_TCP_CONNECTION_TIMEOUT_DURATION_MS,
```

**Review by benpicco — COMMENTED** [review]
This looks good at a cursory glance

File: sys/net/gnrc/transport_layer/tcp/gnrc_tcp.c
Comment (brummer-simon) [reply sys/net/gnrc/transport_layer/tcp/gnrc_tcp.c:510]: Yes. Good catch

File: sys/net/gnrc/transport_layer/tcp/gnrc_tcp.c
Comment (brummer-simon) [reply sys/net/gnrc/transport_layer/tcp/gnrc_tcp.c]: Fixed

File: sys/net/gnrc/transport_layer/tcp/gnrc_tcp_fsm.c
Comment (brummer-simon) [reply sys/net/gnrc/transport_layer/tcp/gnrc_tcp_fsm.c]: Fixed

**Review by benpicco — APPROVED** [review]
Looks good to me, Murdock should ensure that the tests are still passing on native.

**Comment by miri64** [comment]
> Looks good to me, Murdock should ensure that the tests are still passing on native.

Given that the tests require root it probably doesn't ;-).

**Comment by benpicco** [comment]
True - but Murdock found that `compile/fuzzing/gnrc_tcp` does not compile anymore now.

**Comment by brummer-simon** [comment]
> 
> 
> True - but Murdock found that `compile/fuzzing/gnrc_tcp` does not compile anymore now.

Should be fixed by now

**Comment by miri64** [comment]
Nevertheless the `test-as-root` target should be run by a maintainer on a handful of platforms locally before merging ;-). If I find the time I will do :-).

**Review by miri64 — CHANGES_REQUESTED** [review]
Ok, now I looked at the code and found a problem :sweat_smile: .

File: fuzzing/gnrc_tcp/main.c
Comment (miri64) [inline fuzzing/gnrc_tcp/main.c:30]: This could get very large when `TCB_QUEUE_SIZE` is increased and thus increase the stack usage. I don't see any reason why this array has to be allocated on the stack, so maybe make it at least `static` or move it to the global context.

**Review by miri64 — COMMENTED** [review]
Not sure, why my review comment did not get posted though.

File: fuzzing/gnrc_tcp/main.c
Comment (brummer-simon) [reply fuzzing/gnrc_tcp/main.c]: Although I don't think that TCB_QUEUE_SIZE will ever change, I moved it to global context. The reason for the stack allocation was an attept to keep the application the way it was.

**Comment by brummer-simon** [comment]
> 
> 
> Nevertheless the `test-as-root` target should be run by a maintainer on a handful of platforms locally before merging ;-). If I find the time I will do :-).

Thanks

**Comment by miri64** [comment]
Mh test works on `native`, but not on `samr21-xpro`, which isn't the case on master :-/. I'll post the relevant output with echo activated. Maybe you have an idea.

**Comment by miri64** [comment]
<details><summary>Here it is</summary></details>

```
01-conn_lifecycle_as_client.py: success

02-conn_lifecycle_as_server.py: success

03-send_data.py: success

04-receive_data.py: success



make[2]: Nothing to be done for 'all'.
sudo /home/mlenders/Repositories/RIOT-OS/RIOT/dist/tools/ethos/ethos tap0 /dev/ttyACM0 115200 
----> ethos: sending hello.
----> ethos: activating serial pass through.
----> ethos: hello reply received
ifconfig
> > Iface  4  HWaddr: 66:DA:1D:63:7D:CF 
ifconfig
          L2-PDU:1500  MTU:1500  HL:64  Source address length: 6
          Link type: wired
          inet6 addr: fe80::64da:1dff:fe63:7dcf  scope: link  VAL
          inet6 group: ff02::1
          inet6 group: ff02::1:ff63:7dcf
          
> ifconfig
Iface  4  HWaddr: 66:DA:1D:63:7D:CF 
          L2-PDU:1500  MTU:1500  HL:64  Source address length: 6
          Link type: wired
          inet6 addr: fe80::64da:1dff:fe63:7dcf  scope: link  VAL
          inet6 group: ff02::1
          inet6 group: ff02::1:ff63:7dcf
          
> gnrc_tcp_tcb_init
Iface  4  HWaddr: 66:DA:1D:63:7D:CF 
          L2-PDU:1500  MTU:1500  HL:64  Source address length: 6
          Link type: wired
          inet6 addr: fe80::64da:1dff:fe63:7dcf  scope: link  VAL
          inet6 group: ff02::1
          inet6 group: ff02::1:ff63:7dcf
          
> gnrc_tcp_tcb_init: argc=1, argv[0] = gnrc_tcp_tcb_init
gnrc_tcp_open_passive [::]:36511
> gnrc_tcp_open_passive: argc=2, argv[0] = gnrc_tcp_open_passive, argv[1] = [::]:36511
- test_short_payload gnrc_tcp_open_passive: returns 0
gnrc_tcp_recv 1000000 1
> gnrc_tcp_recv: argc=3, argv[0] = gnrc_tcp_recv, argv[1] = 1000000, argv[2] = 1
gnrc_tcp_recv: received 1
pktbuf
> packet buffer: first byte: 0x20002690, last byte: 0x20003e90 (size: 6144)
  position of last byte used: 392
~ unused: 0x20002690 (next: (nil), size: 6144) ~

gnrc_tcp_close



make[2]: Nothing to be done for 'all'.
sudo /home/mlenders/Repositories/RIOT-OS/RIOT/dist/tools/ethos/ethos tap0 /dev/ttyACM0 115200 
----> ethos: sending hello.
----> ethos: activating serial pass through.
----> ethos: hello reply received
ifconfig
> > Iface  4  HWaddr: 66:DA:1D:63:7D:CF 
ifconfig
          L2-PDU:1500  MTU:1500  HL:64  Source address length: 6
          Link type: wired
          inet6 addr: fe80::64da:1dff:fe63:7dcf  scope: link  VAL
          inet6 group: ff02::1
          inet6 group: ff02::1:ff63:7dcf
          
> ifconfig
Iface  4  HWaddr: 66:DA:1D:63:7D:CF 
          L2-PDU:1500  MTU:1500  HL:64  Source address length: 6
          Link type: wired
          inet6 addr: fe80::64da:1dff:fe63:7dcf  scope: link  VAL
          inet6 group: ff02::1
          inet6 group: ff02::1:ff63:7dcf
          
> gnrc_tcp_tcb_init
Iface  4  HWaddr: 66:DA:1D:63:7D:CF 
          L2-PDU:1500  MTU:1500  HL:64  Source address length: 6
          Link type: wired
          inet6 addr: fe80::64da:1dff:fe63:7dcf  scope: link  VAL
          inet6 group: ff02::1
          inet6 group: ff02::1:ff63:7dcf
          
> gnrc_tcp_tcb_init: argc=1, argv[0] = gnrc_tcp_tcb_init
gnrc_tcp_open_passive [::]:44364
> gnrc_tcp_open_passive: argc=2, argv[0] = gnrc_tcp_open_passive, argv[1] = [::]:44364
- test_short_header gnrc_tcp_open_passive: returns 0
pktbuf
> packet buffer: first byte: 0x20002690, last byte: 0x20003e90 (size: 6144)
  position of last byte used: 392
~ unused: 0x20002690 (next: (nil), size: 6144) ~

gnrc_tcp_close



make[2]: Nothing to be done for 'all'.
sudo /home/mlenders/Repositories/RIOT-OS/RIOT/dist/tools/ethos/ethos tap0 /dev/ttyACM0 115200 
----> ethos: sending hello.
----> ethos: activating serial pass through.
----> ethos: hello reply received
ifconfig
> > Iface  4  HWaddr: 66:DA:1D:63:7D:CF 
ifconfig
          L2-PDU:1500  MTU:1500  HL:64  Source address length: 6
          Link type: wired
          inet6 addr: fe80::64da:1dff:fe63:7dcf  scope: link  VAL
          inet6 group: ff02::1
          inet6 group: ff02::1:ff63:7dcf
          
> ifconfig
Iface  4  HWaddr: 66:DA:1D:63:7D:CF 
          L2-PDU:1500  MTU:1500  HL:64  Source address length: 6
          Link type: wired
          inet6 addr: fe80::64da:1dff:fe63:7dcf  scope: link  VAL
          inet6 group: ff02::1
          inet6 group: ff02::1:ff63:7dcf
          
> gnrc_tcp_tcb_init
Iface  4  HWaddr: 66:DA:1D:63:7D:CF 
          L2-PDU:1500  MTU:1500  HL:64  Source address length: 6
          Link type: wired
          inet6 addr: fe80::64da:1dff:fe63:7dcf  scope: link  VAL
          inet6 group: ff02::1
          inet6 group: ff02::1:ff63:7dcf
          
> gnrc_tcp_tcb_init: argc=1, argv[0] = gnrc_tcp_tcb_init
gnrc_tcp_open_passive [::]:38905
> gnrc_tcp_open_passive: argc=2, argv[0] = gnrc_tcp_open_passive, argv[1] = [::]:38905
- test_send_ack_instead_of_syn gnrc_tcp_close

Traceback (most recent call last):
  File "/home/mlenders/Repositories/RIOT-OS/RIOT/tests/gnrc_tcp/tests-as-root/05-garbage-pkts.py", line 154, in <module>
    res = run(test, timeout=10, echo=True)
  File "/home/mlenders/Repositories/RIOT-OS/RIOT/dist/pythonlibs/testrunner/__init__.py", line 30, in run
    testfunc(child)
  File "/home/mlenders/Repositories/RIOT-OS/RIOT/tests/gnrc_tcp/tests-as-root/05-garbage-pkts.py", line 45, in runner
    func(child, tap, host_ll, dst_if, dst_l2, dst_ll, port)
  File "/home/mlenders/Repositories/RIOT-OS/RIOT/tests/gnrc_tcp/tests-as-root/05-garbage-pkts.py", line 124, in test_send_ack_instead_of_syn
    sock.connect(addr_info[0][-1])
socket.timeout: timed out
make: *** [/home/mlenders/Repositories/RIOT-OS/RIOT/makefiles/tests/tests.inc.mk:35: test-as-root] Error 1
```
</details>

**Comment by brummer-simon** [comment]
> 
> Here it is
> 
> ```
> 01-conn_lifecycle_as_client.py: success
> 
> 02-conn_lifecycle_as_server.py: success
> 
> 03-send_data.py: success
> 
> 04-receive_data.py: success
> 
> 
> 
> make[2]: Nothing to be done for 'all'.
> sudo /home/mlenders/Repositories/RIOT-OS/RIOT/dist/tools/ethos/ethos tap0 /dev/ttyACM0 115200 
> ----> ethos: sending hello.
> ----> ethos: activating serial pass through.
> ----> ethos: hello reply received
> ifconfig
> > > Iface  4  HWaddr: 66:DA:1D:63:7D:CF 
> ifconfig
>           L2-PDU:1500  MTU:1500  HL:64  Source address length: 6
>           Link type: wired
>           inet6 addr: fe80::64da:1dff:fe63:7dcf  scope: link  VAL
>           inet6 group: ff02::1
>           inet6 group: ff02::1:ff63:7dcf
>           
> > ifconfig
> Iface  4  HWaddr: 66:DA:1D:63:7D:CF 
>           L2-PDU:1500  MTU:1500  HL:64  Source address length: 6
>           Link type: wired
>           inet6 addr: fe80::64da:1dff:fe63:7dcf  scope: link  VAL
>           inet6 group: ff02::1
>           inet6 group: ff02::1:ff63:7dcf
>           
> > gnrc_tcp_tcb_init
> Iface  4  HWaddr: 66:DA:1D:63:7D:CF 
>           L2-PDU:1500  MTU:1500  HL:64  Source address length: 6
>           Link type: wired
>           inet6 addr: fe80::64da:1dff:fe63:7dcf  scope: link  VAL
>           inet6 group: ff02::1
>           inet6 group: ff02::1:ff63:7dcf
>           
> > gnrc_tcp_tcb_init: argc=1, argv[0] = gnrc_tcp_tcb_init
> gnrc_tcp_open_passive [::]:36511
> > gnrc_tcp_open_passive: argc=2, argv[0] = gnrc_tcp_open_passive, argv[1] = [::]:36511
> - test_short_payload gnrc_tcp_open_passive: returns 0
> gnrc_tcp_recv 1000000 1
> > gnrc_tcp_recv: argc=3, argv[0] = gnrc_tcp_recv, argv[1] = 1000000, argv[2] = 1
> gnrc_tcp_recv: received 1
> pktbuf
> > packet buffer: first byte: 0x20002690, last byte: 0x20003e90 (size: 6144)
>   position of last byte used: 392
> ~ unused: 0x20002690 (next: (nil), size: 6144) ~
> 
> gnrc_tcp_close
> 
> 
> 
> make[2]: Nothing to be done for 'all'.
> sudo /home/mlenders/Repositories/RIOT-OS/RIOT/dist/tools/ethos/ethos tap0 /dev/ttyACM0 115200 
> ----> ethos: sending hello.
> ----> ethos: activating serial pass through.
> ----> ethos: hello reply received
> ifconfig
> > > Iface  4  HWaddr: 66:DA:1D:63:7D:CF 
> ifconfig
>           L2-PDU:1500  MTU:1500  HL:64  Source address length: 6
>           Link type: wired
>           inet6 addr: fe80::64da:1dff:fe63:7dcf  scope: link  VAL
>           inet6 group: ff02::1
>           inet6 group: ff02::1:ff63:7dcf
>           
> > ifconfig
> Iface  4  HWaddr: 66:DA:1D:63:7D:CF 
>           L2-PDU:1500  MTU:1500  HL:64  Source address length: 6
>           Link type: wired
>           inet6 addr: fe80::64da:1dff:fe63:7dcf  scope: link  VAL
>           inet6 group: ff02::1
>           inet6 group: ff02::1:ff63:7dcf
>           
> > gnrc_tcp_tcb_init
> Iface  4  HWaddr: 66:DA:1D:63:7D:CF 
>           L2-PDU:1500  MTU:1500  HL:64  Source address length: 6
>           Link type: wired
>           inet6 addr: fe80::64da:1dff:fe63:7dcf  scope: link  VAL
>           inet6 group: ff02::1
>           inet6 group: ff02::1:ff63:7dcf
>           
> > gnrc_tcp_tcb_init: argc=1, argv[0] = gnrc_tcp_tcb_init
> gnrc_tcp_open_passive [::]:44364
> > gnrc_tcp_open_passive: argc=2, argv[0] = gnrc_tcp_open_passive, argv[1] = [::]:44364
> - test_short_header gnrc_tcp_open_passive: returns 0
> pktbuf
> > packet buffer: first byte: 0x20002690, last byte: 0x20003e90 (size: 6144)
>   position of last byte used: 392
> ~ unused: 0x20002690 (next: (nil), size: 6144) ~
> 
> gnrc_tcp_close
> 
> 
> 
> make[2]: Nothing to be done for 'all'.
> sudo /home/mlenders/Repositories/RIOT-OS/RIOT/dist/tools/ethos/ethos tap0 /dev/ttyACM0 115200 
> ----> ethos: sending hello.
> ----> ethos: activating serial pass through.
> ----> ethos: hello reply received
> ifconfig
> > > Iface  4  HWaddr: 66:DA:1D:63:7D:CF 
> ifconfig
>           L2-PDU:1500  MTU:1500  HL:64  Source address length: 6
>           Link type: wired
>           inet6 addr: fe80::64da:1dff:fe63:7dcf  scope: link  VAL
>           inet6 group: ff02::1
>           inet6 group: ff02::1:ff63:7dcf
>           
> > ifconfig
> Iface  4  HWaddr: 66:DA:1D:63:7D:CF 
>           L2-PDU:1500  MTU:1500  HL:64  Source address length: 6
>           Link type: wired
>           inet6 addr: fe80::64da:1dff:fe63:7dcf  scope: link  VAL
>           inet6 group: ff02::1
>           inet6 group: ff02::1:ff63:7dcf
>           
> > gnrc_tcp_tcb_init
> Iface  4  HWaddr: 66:DA:1D:63:7D:CF 
>           L2-PDU:1500  MTU:1500  HL:64  Source address length: 6
>           Link type: wired
>           inet6 addr: fe80::64da:1dff:fe63:7dcf  scope: link  VAL
>           inet6 group: ff02::1
>           inet6 group: ff02::1:ff63:7dcf
>           
> > gnrc_tcp_tcb_init: argc=1, argv[0] = gnrc_tcp_tcb_init
> gnrc_tcp_open_passive [::]:38905
> > gnrc_tcp_open_passive: argc=2, argv[0] = gnrc_tcp_open_passive, argv[1] = [::]:38905
> - test_send_ack_instead_of_syn gnrc_tcp_close
> 
> Traceback (most recent call last):
>   File "/home/mlenders/Repositories/RIOT-OS/RIOT/tests/gnrc_tcp/tests-as-root/05-garbage-pkts.py", line 154, in <module>
>     res = run(test, timeout=10, echo=True)
>   File "/home/mlenders/Repositories/RIOT-OS/RIOT/dist/pythonlibs/testrunner/__init__.py", line 30, in run
>     testfunc(child)
>   File "/home/mlenders/Repositories/RIOT-OS/RIOT/tests/gnrc_tcp/tests-as-root/05-garbage-pkts.py", line 45, in runner
>     func(child, tap, host_ll, dst_if, dst_l2, dst_ll, port)
>   File "/home/mlenders/Repositories/RIOT-OS/RIOT/tests/gnrc_tcp/tests-as-root/05-garbage-pkts.py", line 124, in test_send_ack_instead_of_syn
>     sock.connect(addr_info[0][-1])
> socket.timeout: timed out
> make: *** [/home/mlenders/Repositories/RIOT-OS/RIOT/makefiles/tests/tests.inc.mk:35: test-as-root] Error 1
> ```

The log seems strange to me since it sends gnrc_tcp_open_passive to the target, a function that should not exist anymore. Could you perform a clean build and test again?

**Comment by miri64** [comment]
I don't see how a clean build has any effect on the test script ;-).

**Comment by miri64** [comment]
But did just that and same result.

**Comment by miri64** [comment]
Ah wait, the output indeed differs

<details>

```
01-conn_lifecycle_as_client.py: success


- Test "test_lifecycle_as_server": SUCCESS
02-conn_lifecycle_as_server.py: success

03-send_data.py: success

04-receive_data.py: success



make[2]: Nothing to be done for 'all'.
sudo /home/mlenders/Repositories/RIOT-OS/RIOT/dist/tools/ethos/ethos tap0 /dev/ttyACM0 115200 
----> ethos: sending hello.
----> ethos: activating serial pass through.
----> ethos: hello reply received
ifconfig
> > Iface  4  HWaddr: 66:DA:1D:63:7D:CF 
ifconfig
          L2-PDU:1500  MTU:1500  HL:64  Source address length: 6
          Link type: wired
          inet6 addr: fe80::64da:1dff:fe63:7dcf  scope: link  VAL
          inet6 group: ff02::1
          inet6 group: ff02::1:ff63:7dcf
          
> ifconfig
Iface  4  HWaddr: 66:DA:1D:63:7D:CF 
          L2-PDU:1500  MTU:1500  HL:64  Source address length: 6
          Link type: wired
          inet6 addr: fe80::64da:1dff:fe63:7dcf  scope: link  VAL
          inet6 group: ff02::1
          inet6 group: ff02::1:ff63:7dcf
          
> gnrc_tcp_tcb_init
Iface  4  HWaddr: 66:DA:1D:63:7D:CF 
          L2-PDU:1500  MTU:1500  HL:64  Source address length: 6
          Link type: wired
          inet6 addr: fe80::64da:1dff:fe63:7dcf  scope: link  VAL
          inet6 group: ff02::1
          inet6 group: ff02::1:ff63:7dcf
          
> gnrc_tcp_tcb_init: argc=1, argv[0] = gnrc_tcp_tcb_init
gnrc_tcp_listen [::]:20132
gnrc_tcp_tcb_init: returns 0
> gnrc_tcp_listen: argc=2, argv[0] = gnrc_tcp_listen, argv[1] = [::]:20132
gnrc_tcp_accept 2000
- test_short_payload gnrc_tcp_listen: returns 0
> gnrc_tcp_accept: argc=2, argv[0] = gnrc_tcp_accept, argv[1] = 2000
gnrc_tcp_accept: returns 0
gnrc_tcp_recv 1000000 1
> gnrc_tcp_recv: argc=3, argv[0] = gnrc_tcp_recv, argv[1] = 1000000, argv[2] = 1
gnrc_tcp_recv: received 1
pktbuf
> packet buffer: first byte: 0x200026b4, last byte: 0x20003eb4 (size: 6144)
  position of last byte used: 392
~ unused: 0x200026b4 (next: (nil), size: 6144) ~

gnrc_tcp_close
gnrc_tcp_stop_listen



make[2]: Nothing to be done for 'all'.
sudo /home/mlenders/Repositories/RIOT-OS/RIOT/dist/tools/ethos/ethos tap0 /dev/ttyACM0 115200 
----> ethos: sending hello.
----> ethos: activating serial pass through.
----> ethos: hello reply received
ifconfig
> > Iface  4  HWaddr: 66:DA:1D:63:7D:CF 
ifconfig
          L2-PDU:1500  MTU:1500  HL:64  Source address length: 6
          Link type: wired
          inet6 addr: fe80::64da:1dff:fe63:7dcf  scope: link  VAL
          inet6 group: ff02::1
          inet6 group: ff02::1:ff63:7dcf
          
> ifconfig
Iface  4  HWaddr: 66:DA:1D:63:7D:CF 
          L2-PDU:1500  MTU:1500  HL:64  Source address length: 6
          Link type: wired
          inet6 addr: fe80::64da:1dff:fe63:7dcf  scope: link  VAL
          inet6 group: ff02::1
          inet6 group: ff02::1:ff63:7dcf
          
> gnrc_tcp_tcb_init
Iface  4  HWaddr: 66:DA:1D:63:7D:CF 
          L2-PDU:1500  MTU:1500  HL:64  Source address length: 6
          Link type: wired
          inet6 addr: fe80::64da:1dff:fe63:7dcf  scope: link  VAL
          inet6 group: ff02::1
          inet6 group: ff02::1:ff63:7dcf
          
> gnrc_tcp_tcb_init: argc=1, argv[0] = gnrc_tcp_tcb_init
gnrc_tcp_listen [::]:41242
gnrc_tcp_tcb_init: returns 0
> gnrc_tcp_listen: argc=2, argv[0] = gnrc_tcp_listen, argv[1] = [::]:41242
gnrc_tcp_accept 2000
- test_short_header gnrc_tcp_listen: returns 0
> gnrc_tcp_accept: argc=2, argv[0] = gnrc_tcp_accept, argv[1] = 2000
gnrc_tcp_accept: returns 0
pktbuf
> packet buffer: first byte: 0x200026b4, last byte: 0x20003eb4 (size: 6144)
  position of last byte used: 392
~ unused: 0x200026b4 (next: (nil), size: 6144) ~

gnrc_tcp_close
gnrc_tcp_stop_listen



make[2]: Nothing to be done for 'all'.
sudo /home/mlenders/Repositories/RIOT-OS/RIOT/dist/tools/ethos/ethos tap0 /dev/ttyACM0 115200 
----> ethos: sending hello.
----> ethos: activating serial pass through.
----> ethos: hello reply received
ifconfig
> > Iface  4  HWaddr: 66:DA:1D:63:7D:CF 
ifconfig
          L2-PDU:1500  MTU:1500  HL:64  Source address length: 6
          Link type: wired
          inet6 addr: fe80::64da:1dff:fe63:7dcf  scope: link  VAL
          inet6 group: ff02::1
          inet6 group: ff02::1:ff63:7dcf
          
> ifconfig
Iface  4  HWaddr: 66:DA:1D:63:7D:CF 
          L2-PDU:1500  MTU:1500  HL:64  Source address length: 6
          Link type: wired
          inet6 addr: fe80::64da:1dff:fe63:7dcf  scope: link  VAL
          inet6 group: ff02::1
          inet6 group: ff02::1:ff63:7dcf
          
> gnrc_tcp_tcb_init
Iface  4  HWaddr: 66:DA:1D:63:7D:CF 
          L2-PDU:1500  MTU:1500  HL:64  Source address length: 6
          Link type: wired
          inet6 addr: fe80::64da:1dff:fe63:7dcf  scope: link  VAL
          inet6 group: ff02::1
          inet6 group: ff02::1:ff63:7dcf
          
> gnrc_tcp_tcb_init: argc=1, argv[0] = gnrc_tcp_tcb_init
gnrc_tcp_listen [::]:51203
gnrc_tcp_tcb_init: returns 0
> gnrc_tcp_listen: argc=2, argv[0] = gnrc_tcp_listen, argv[1] = [::]:51203
gnrc_tcp_accept 2000
- test_send_ack_instead_of_syn gnrc_tcp_listen: returns 0
> gnrc_tcp_accept: argc=2, argv[0] = gnrc_tcp_accept, argv[1] = 2000
gnrc_tcp_accept: returns -ETIMEDOUT
> gnrc_tcp_close
gnrc_tcp_stop_listen
```

</details>

**Comment by brummer-simon** [comment]
Seems like a timeout thats large enough on native. I'll see if I can find a board here and try to determine a suitable value

**Comment by brummer-simon** [comment]
@miri64  - I tested the PR on the nucleo-f401re and it worked. I added a commit increasing the accept timeout in the failing test script from 2000ms to 4000ms, I hope this is long enough for all boards with less processing power.

The failing test is the one where 1000 malformed packets are send and discarded, afterwards a valid packet is sent and used to accept the connection. In that test processing power might be the limiting factor. Could you test again?

**Comment by miri64** [comment]
Nope. But 10000 works. How about, just to be on the save side to set it to 15000? Also if that is done, please squash the commits reflecting the review history.

**Comment by brummer-simon** [comment]
> Nope. But 10000 works. How about, just to be on the save side to set it to 15000? Also if that is done, please squash the commits reflecting the review history.

Done

**Comment by miri64** [comment]
0d5db28 still needs to be squashed in the original commits.

**Comment by brummer-simon** [comment]
> [0d5db28](https://github.com/RIOT-OS/RIOT/commit/0d5db28c4082464395615db6aea88a1535c903da) still needs to be squashed in the original commits.

Forgot this one. I squashed it.

**Review by miri64 — APPROVED** [review]
ACK

**Comment by miri64** [comment]
Not sure why murdock is marked as failed or still doing label checks...

**Comment by brummer-simon** [comment]
@miri64: Should I squash it in a single commit? I splitted the PR only to help with the review.

**Comment by brummer-simon** [comment]
@miri64 , @benpicco, since everything is green, could you press the merge button? For an unknown reason I am not able to do that anymore :(

**Comment by miri64** [comment]
> @miri64 , @benpicco, since everything is green, could you press the merge button? For an unknown reason I am not able to do that anymore :(

Where you ever? You need write permissions to hit the green button (which only maintainers have).

**Comment by brummer-simon** [comment]
Maybe my mind is playing a trick on me but I could swere that there was I time were I was able to press the merge button for my own PR after a successfull review. But I had never maintainer status.

**Pull request merged**
